src/main/webapp/WEB-INF/view/searchbar.jsp

```diff
@@ -1,45 +1,44 @@
-<%@ page import="codeu.model.data.User" %>
-<%@ page import="codeu.model.store.basic.UserStore" %>
-<%@ page import="java.util.List" %>
-<%
-List<User> allUsers = UserStore.getInstance().getAllUsers();
-%>
-
 <script type="text/javascript">
-  // Checks if the search request is empty.
-  function isEmpty() {
-    var searchBar = document.getElementById("searchBar");
-    if(searchBar.value.length < 1) {
-      alert("Please enter a valid search.");
-      return false;
-    }
-    return true;
-  }
-
-  // Toggle drop down.
-  function toggleDropDown() {
-    var searchBar = document.getElementById("searchBar");
-    var dropDiv = document.getElementById("dropDiv");
-    if (searchBar.value == "") {
-      dropDiv.style.display = "none";
-    }
-    else {
-      dropDiv.style.display = "block"
+  // Iterates through fetched results
+  function fetchResults() {
+    // Clear the resultItem divs
+    document.querySelector('#result').innerHTML = '';
+    var searchBarValue = document.querySelector('#searchBar').value;
+    if (searchBarValue.length > 0) {
+      fetch('/search?searchRequest=' + searchBarValue, 
+        {credentials: "same-origin"})
+      .then(
+        function(response) {
+          if (response.status !== 200) {
+            console.log("Looks like there was a problem. Status Code: " + response.status);
+            return;
+          }
+          response.json().then(function(data) {
+            if (data != null) {
```

Comment:
nit: this `if` is redundant, the for loop below won't execute if data is null.

Reply:
Thanks for pointing this out